### ISSUE TITLE
Waypoint RVIZ Gui problem

### ISSUE THREAD

**Issue by cburbridge** [strands-project/strands_navigation#46]
After #43, when I try to move a waypoint in RViz, I get:

```
[ERROR] [WallTime: 1399280078.407084] bad callback: <bound method InteractiveMarkerServer.processFeedback of <interactive_markers.interactive_marker_server.InteractiveMarkerServer instance at 0x2aec050>>
Traceback (most recent call last):
  File "/opt/ros/hydro/lib/python2.7/dist-packages/rospy/topics.py", line 682, in _invoke_callback
    cb(msg)
  File "/opt/ros/hydro/lib/python2.7/dist-packages/interactive_markers/interactive_marker_server.py", line 356, in processFeedback
    marker_context.default_feedback_cb(feedback)
  File "/opt/strands/strands_catkin_ws/src/strands_navigation/topological_utils/scripts/visualise_map.py", line 292, in _marker_feedback
    self.topo_map.update_node_waypoint(feedback.marker_name, feedback.pose)
  File "/opt/strands/strands_catkin_ws/src/strands_navigation/topological_navigation/src/topological_navigation/topological_map.py", line 40, in update_node_waypoint
    available[0][0].pose = new_pose
IndexError: list index out of range
```

The way point does not update.

**Comment by Jailander** [comment]
Hi, I can't reproduce this error, can you try merging the next pull request https://github.com/strands-project/strands_navigation/pull/45 and see if that solves the problem?

**Comment by Jailander** [comment]
Be careful merging that pull request means that will need to run:
`rosrun topological_utils map_collection_change.py` once to move the topological maps to a new collection called topological maps

**Comment by cburbridge** [comment]
Still the same error...

**Comment by Jailander** [comment]
OK, I am looking into it, it may be connected with the bug you reported in #45 I will commit a small fix for that

**Comment by cburbridge** [comment]
the bug in #45 only appeared when i had forgotten to run the update script.

**Comment by Jailander** [comment]
Can you send me the JSON of a node in the topological maps collection on robomongo or similar, I still can't reproduce the error :/

**Comment by cburbridge** [comment]
http://pastebin.com/wKpsndSe

**Comment by Jailander** [comment]
Ok, the nodes seem to be well defined in your datacentre, is it  possible that there is a problem with different versions of the ros_datacentre? I have tested this after updating my system (an hour ago) and it works fine, I also tested on Christians PC (with a fresh datacentre) and it works fine as well. Can you try on another system?

**Comment by cburbridge** [comment]
I get the same error an updated system, both on my laptop in simulation and
also on bob.

**Comment by Jailander** [comment]
Ok this is confusing :/ can you try this new pull request that I just created and tell me what it says now there should be a warning, I think is the only way I can shed some light on the problem

**Comment by cburbridge** [comment]
Awesome, now it just says 
`INFO] [WallTime: 1399381259.829408] Waiting for services...
[INFO] [WallTime: 1399381259.844581] Done
`
and presumably works. How can I delete edges?

**Comment by Jailander** [comment]
Ok, just to make sure, could you move the waypoints around (wait until the screen refreshes that means it was stored on the datacentre) and restarting the script, the waypoints should remain where you left them last.

**Comment by Jailander** [comment]
Sorry, I forgot to tell you about the edges, I am currently tiding the scripts a little i will commit that soon, meanwhile you will have to keep using robomongo sorry, I promise it will be done in an hour or so in any case.

**Comment by cburbridge** [comment]
No worries!
